### ISSUE TITLE
New Lego Reader Not working

### ISSUE THREAD

**Issue by agrignard** [CityScope/CSL_Lima#29]
@JesusGarcia98 

Can you explain a bit more all the change made in this commit (I am sure it is a nice improvement however it doesn't work as it was)
https://github.com/CityScope/CSL_Lima/commit/da4e27fc290b546847a8a5c8218de2db776bcf3a

I had a demo running with the last version of the LegoReader and after updating with the new LegoReader it doesn't work anymore on my demo and I notice different bug while running it

Would it be possible to revert those change and  make this change in a separate branch until it is stable so it does not impact the master branch

Usually any change on the master branch should be trigger by a specific issue.

We have to be careful , especially in an Open Source projet shared by different people to be sure that change that are made do not affect the rest of the user. 

We should keep on using the pull request feature to avoid this kind of issue

**Comment by JesusGarcia98** [comment]
Hello!

The commit you mention helped optimizing the LegoReader, QRreader and HumanitarianModel programs. In general, unnecessary loops, redundant pieces of code and some bad practices in programming were removed.  More specifically, for LegoReader the CPU usage was brought down from 250% to around 140%, for QRreader though I do not remember its previous usage of CPU I am sure it diminished and lastly, for HumanitarianModel the loading method was remade and now it takes considerably less time to start.

I created a new branch called 'dev', where I committed a fix for the new version of LegoReader. Also, the previous version you had can now be found in the master branch.

The changes made to LegoReader were to acknowledge the glaring issue of the unreasonably high CPU usage. However, I will be more careful in the future to avoid affecting the rest of users.

About the new bug you find in the console when running LegoReader, I investigated about it, but it seems to be some kind of internal debugging feature for OpenGL which only happens in Mac.

**Comment by agrignard** [comment]
Hi @JesusGarcia98 Thank you for the very quick and efficient solution.

This is a great workflow to have the "stable version" in the master on the "enhance still in dev" in the dev branch. For now the demo in the lab works with the last master branch. I will report the issue that I found in the dev (I'll report specific issues for each of them). From what I tested so far it seems that the dev version is much more faster so it's maybe so small adjustement.
